From 1d7145ef6a6cf1827b724586bc7aa2eb479e7f1c Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 29 Jan 2017 04:50:15 +0000
Subject: [PATCH] panel: Restore missing animations (GTK >= 3.22)

This is an emergency fix and should be reverted by maintainers using GTK
versions older than 3.22. Due to our use of autotools, it ships the .c files
in the tarball automatically, rendering any use of compiler conditionals to
be useless.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/panel/applets/budgie-menu/BudgieMenu.vala               |  6 +++---
 src/panel/applets/clock/ClockApplet.vala                    |  2 +-
 src/panel/applets/keyboard-layout/KeyboardLayoutApplet.vala |  4 ++--
 src/panel/applets/places-indicator/PlacesIndicator.vala     |  4 ++--
 src/panel/applets/status/BluetoothIndicator.vala            |  2 +-
 src/panel/applets/status/StatusApplet.vala                  |  2 +-
 src/panel/applets/user-indicator/UserIndicator.vala         |  2 +-
 src/panel/popovers.vala                                     | 10 +++++-----
 8 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/src/panel/applets/budgie-menu/BudgieMenu.vala b/src/panel/applets/budgie-menu/BudgieMenu.vala
index 1887a2e..c711d69 100644
--- a/src/panel/applets/budgie-menu/BudgieMenu.vala
+++ b/src/panel/applets/budgie-menu/BudgieMenu.vala
@@ -110,7 +110,7 @@ public class BudgieMenuApplet : Budgie.Applet
                 return Gdk.EVENT_PROPAGATE;
             }
             if (popover.get_visible()) {
-                popover.hide();
+                popover.popdown();
             } else {
                 popover.get_child().show_all();
                 this.manager.show_popover(widget);
@@ -137,7 +137,7 @@ public class BudgieMenuApplet : Budgie.Applet
 
         popover.key_release_event.connect((e)=> {
             if (e.keyval == Gdk.Key.Escape) {
-                popover.hide();
+                popover.popdown();
             }
             return Gdk.EVENT_PROPAGATE;
         });
@@ -147,7 +147,7 @@ public class BudgieMenuApplet : Budgie.Applet
     {
         if ((action & Budgie.PanelAction.MENU) != 0) {
             if (popover.get_visible()) {
-                popover.hide();
+                popover.popdown();
             } else {
                 popover.get_child().show_all();
                 this.manager.show_popover(widget);
diff --git a/src/panel/applets/clock/ClockApplet.vala b/src/panel/applets/clock/ClockApplet.vala
index a0bef75..e9a5c66 100644
--- a/src/panel/applets/clock/ClockApplet.vala
+++ b/src/panel/applets/clock/ClockApplet.vala
@@ -63,7 +63,7 @@ public class ClockApplet : Budgie.Applet
                 return Gdk.EVENT_PROPAGATE;
             }
             if (popover.get_visible()) {
-                popover.hide();
+                popover.popdown();
             } else {
                 this.manager.show_popover(widget);
             }
diff --git a/src/panel/applets/keyboard-layout/KeyboardLayoutApplet.vala b/src/panel/applets/keyboard-layout/KeyboardLayoutApplet.vala
index 982f5fa..fa1945b 100644
--- a/src/panel/applets/keyboard-layout/KeyboardLayoutApplet.vala
+++ b/src/panel/applets/keyboard-layout/KeyboardLayoutApplet.vala
@@ -261,7 +261,7 @@ public class KeyboardLayoutApplet : Budgie.Applet
                 return Gdk.EVENT_PROPAGATE;
             }
             if (popover.get_visible()) {
-                popover.hide();
+                popover.popdown();
             } else {
                 this.manager.show_popover(widget);
             }
@@ -333,7 +333,7 @@ public class KeyboardLayoutApplet : Budgie.Applet
         uint idx = btn.idx;
 
         this.settings.set_uint("current", idx);
-        popover.hide();
+        popover.popdown();
     }
 
     /**
diff --git a/src/panel/applets/places-indicator/PlacesIndicator.vala b/src/panel/applets/places-indicator/PlacesIndicator.vala
index b5f3ceb..120ca20 100644
--- a/src/panel/applets/places-indicator/PlacesIndicator.vala
+++ b/src/panel/applets/places-indicator/PlacesIndicator.vala
@@ -106,7 +106,7 @@ public class PlacesIndicatorApplet : Budgie.Applet
     public void toggle_popover()
     {
         if (popover.get_visible()) {
-            popover.hide();
+            popover.popdown();
         } else {
             popover.get_child().show_all();
             this.manager.show_popover(ebox);
@@ -151,4 +151,4 @@ public void peas_register_types(TypeModule module)
     // boilerplate - all modules need this
     var objmodule = module as Peas.ObjectModule;
     objmodule.register_extension_type(typeof(Budgie.Plugin), typeof(PlacesIndicator));
-}
\ No newline at end of file
+}
diff --git a/src/panel/applets/status/BluetoothIndicator.vala b/src/panel/applets/status/BluetoothIndicator.vala
index 3487467..7aa751f 100644
--- a/src/panel/applets/status/BluetoothIndicator.vala
+++ b/src/panel/applets/status/BluetoothIndicator.vala
@@ -208,7 +208,7 @@ public class BluetoothIndicator : Gtk.Bin
         } catch (Error e) {
             message("Error setting airplane mode: %s", e.message);
         }
-        this.popover.hide();
+        this.popover.popdown();
     }
 
     /* Notify */
diff --git a/src/panel/applets/status/StatusApplet.vala b/src/panel/applets/status/StatusApplet.vala
index c1f7ccc..0619545 100644
--- a/src/panel/applets/status/StatusApplet.vala
+++ b/src/panel/applets/status/StatusApplet.vala
@@ -36,7 +36,7 @@ public class StatusApplet : Budgie.Applet
                 return Gdk.EVENT_PROPAGATE;
             }
             if (popover.get_visible()) {
-                popover.hide();
+                popover.popdown();
             } else {
                 this.manager.show_popover(parent_widget);
             }
diff --git a/src/panel/applets/user-indicator/UserIndicator.vala b/src/panel/applets/user-indicator/UserIndicator.vala
index 6bcfae2..6593466 100644
--- a/src/panel/applets/user-indicator/UserIndicator.vala
+++ b/src/panel/applets/user-indicator/UserIndicator.vala
@@ -53,7 +53,7 @@ public class UserIndicatorApplet : Budgie.Applet {
     
     public void Toggle(){
         if (popover.get_visible()) {
-            popover.hide();
+            popover.popdown();
         } else {
             popover.get_child().show_all();
             this.manager.show_popover(ebox);
diff --git a/src/panel/popovers.vala b/src/panel/popovers.vala
index 1245b18..8d513f9 100644
--- a/src/panel/popovers.vala
+++ b/src/panel/popovers.vala
@@ -33,7 +33,7 @@ public class PopoverManagerImpl : PopoverManager, GLib.Object
                 return Gdk.EVENT_PROPAGATE;
             }
             if (visible_popover != null) {
-                visible_popover.hide();
+                visible_popover.popdown();
                 make_modal(visible_popover, false);
                 visible_popover = null;
             }
@@ -54,7 +54,7 @@ public class PopoverManagerImpl : PopoverManager, GLib.Object
             }
             if ((e.x < alloc.x || e.x > alloc.x+alloc.width) ||
                 (e.y < alloc.y || e.y > alloc.y+alloc.height)) {
-                    visible_popover.hide();
+                    visible_popover.popdown();
                     make_modal(visible_popover, false);
                     visible_popover = null;
             }
@@ -102,7 +102,7 @@ public class PopoverManagerImpl : PopoverManager, GLib.Object
         }
 
         owner.set_expanded(true);
-        pop.show();
+        pop.popup();
     }
 
     void on_widget_mapped(Gtk.Widget? p)
@@ -130,9 +130,9 @@ public class PopoverManagerImpl : PopoverManager, GLib.Object
                 if (visible_popover != widgets[w] && visible_popover != null) {
                     /* Hide current popover, re-open next */
                     mousing = true;
-                    visible_popover.hide();
+                    visible_popover.popdown();
                     visible_popover = widgets[w];
-                    visible_popover.show_all();
+                    visible_popover.popup();
                     owner.set_focus(null);
                     visible_popover.grab_focus();
                     mousing = false;
-- 
2.11.0

